pyqtorch/primitive.py

```diff
@@ -200,9 +195,28 @@ def __init__(self, target: int):
 
 class SWAP(Primitive):
     def __init__(self, control: int, target: int):
+        # TODO: Change the control param name
         super().__init__(OPERATIONS_DICT["SWAP"], target)
-        self.control = (control,) if isinstance(control, int) else control
-        self.qubit_support = self.control + (target,)
+        self.control = control
+        self.qubit_support = (self.control,) + (target,)
+
+    def tensor(
+        self, values: dict[str, Tensor] = {}, n_qubits: int = 1, diagonal: bool = False
+    ) -> Tensor:
+        from pyqtorch.circuit import Sequence
+
+        if diagonal:
+            raise NotImplementedError
+        if n_qubits < max(self.qubit_support) + 1:
+            n_qubits = max(self.qubit_support) + 1
+        seq = Sequence(
+            [
+                CNOT(control=self.control, target=self.target),  # type: ignore[arg-type]
+                CNOT(control=self.target, target=self.control),  # type: ignore[arg-type]
+                CNOT(control=self.control, target=self.target),  # type: ignore[arg-type]
+            ]
+        )
+        return seq.tensor(values, n_qubits)
```

Comment:
Why is this needed, is there not a SWAP matrix stored that can be used directly?